File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.js
function solution(n) {
    let f = 1;
    let mul = 1;
    
    for(f = 1; f <= n; f++) {
        if(mul * f <= n) {
            mul *= f;
        } else {
            break;
        }
    }
    return --f;
}

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.py
import math

def solution(n):
    fact = []
    delta = []
    
    for i in range(1, 11):
        fact.append(math.factorial(i))

    for f in fact:
        if n - f >= 0:
            delta.append(n - f)
    
    return delta.index(min(delta)) + 1